### PR TITLE
Add validation to move in frontend

### PR DESCRIPTION
**Pull request by Ojself**
When player clicks a tile and wants to make a move, the move is being validated and sent to backend for further validation. 

An error message is displayed for the user if the request goes into catch

### CODE REVIEW

**Comment by Parnswir** [comment]
What exactly are the problems, except that there's no answer from the server yet?

**Comment by Ojself** [comment]
> What exactly are the problems, except that there's no answer from the server yet?

There's no obvious problem to be honest. The code runs as expected. But I don't know if the logic is as clever as we want it to be. Maybe there's a better approach.

File: routes/api/index.js [hunk 1]
```diff
@@ -10,8 +10,16 @@ router.get('/board', (req, res) => {
...
+	const foo = req.body.body;
+	return res.json({ foo });
+});
+
+// move made by user are sent here, verified and updated
+router.post('/move', (req, res) => {
+	console.log(req.body, "here's your data");
+	// Verify move
+	// const newBoard = createNewGame(req.body)
+	// res.render('board', newBoard)
```
Comment (Parnswir) [inline routes/api/index.js:22]: I'm working on this in parallel. I'd suggest this until I get around to actually implementing the move. This way, no matter what move you make, the default board is returned, so you can think about re-rendering and stuff :-)
```suggestion
	return res.json(BoardFactory.toObject(new Board());
```

File: public/shapes/disc.js [hunk 2]
```diff
@@ -21,6 +21,9 @@ class Disc {
 
 	draw() {
 		fill('#e5e5e5');
+		/* if (this.hovering) {
+			fill('#7fff00');
+		} */
```
Comment (Parnswir) [inline public/shapes/disc.js:26]:
```suggestion
		
```

File: public/scripts/game.js [hunk 3]
```diff
@@ -78,23 +85,55 @@ class Game {
...
+				})
+				.catch((error) => {
+					console.log(error);
+					this.errorMsg = error;
+
+					// draws error message in nice color
+					setTimeout(() => {
+						this.errorMsg = null;
+					}, 5000);
+				});
```
Comment (Parnswir) [inline public/scripts/game.js:134]: I bet you can extract this to be a more generic error handler, that we can re-use for all the other requests (and to clean up the code a bit)

File: public/scripts/game.js [hunk 4]
```diff
@@ -41,12 +43,17 @@ class Game {
...
 			disc.draw();
 		});
 		this.tokens.forEach((token) => {
-			token.hovering = this.hoverCheck(token, mouseX, mouseY);
+			token.hovering = this.tileHoverCheck(token, mouseX, mouseY);
 			token.draw();
 		});
+		if (this.errorMsg) {
+			fill('tomato');
+			text(`${JSON.stringify(this.errorMsg)}`, 25, 150);
```
Comment (Parnswir) [inline public/scripts/game.js:55]: Why use `JSON.stringify` here?

**Review by Parnswir — CHANGES_REQUESTED** [review]

File: public/scripts/game.js (hunk 4 above)
Comment (Ojself) [reply public/scripts/game.js]: Good question. I removed it!

**Review by Parnswir — APPROVED** [review]

**Pull request merged**